src/components/MovieForm/MovieForm.tsx

```diff
@@ -0,0 +1,364 @@
+            placeholder="Title"
+            action={FormActionType.UPDATE_TITLE}
+            showValidation={showHintsState && !titleState.isValid}
+            validationMessage={dict.FORM_TITLE_VALIDATION}
+          />
+          <MovieFormInput
+            title={constants.SORTOPTIONS.releasedate}
+            id="releaseDate"
+            type="date"
+            value={dateState.value}
+            dispatch={dispatchForm}
+            placeholder=""
+            action={FormActionType.UPDATE_DATE}
+            showValidation={showHintsState && !dateState.isValid}
+            validationMessage={dict.FORM_DATE_VALIDATION}
+          />
+          <MovieFormInput
+            title={dict.FORM_URL}
+            id="url"
+            type="text"
+            value={urlState.value}
+            dispatch={dispatchForm}
+            placeholder="https://"
+            action={FormActionType.UPDATE_URL}
+            showValidation={showHintsState && !urlState.isValid}
+            validationMessage={dict.FORM_URL_VALIDATION}
+          />
+          <MovieFormInput
+            title={constants.SORTOPTIONS.rating}
+            id="rating"
+            type="text"
+            value={ratingState.value}
+            dispatch={dispatchForm}
+            placeholder="7.8"
+            action={FormActionType.UPDATE_RATING}
+            showValidation={showHintsState && !ratingState.isValid}
+            validationMessage={dict.FORM_RATING_VALIDATION}
+          />
+          <GenreDropdown
+            title={dict.GENRE}
+            id="genres"
+            showValidation={showHintsState && !genreState.isValid}
+            validationMessage={dict.FORM_GENRE_VALIDATION}
+            placeholder={dict.FORM_GENRE_PLACEHOLDER}
+            genres={possibleGenres}
+            value={genreState.value}
+            dispatch={dispatchForm}
+            action={FormActionType.UPDATE_GENRE}
+          />
+          <MovieFormInput
+            title={constants.SORTOPTIONS.runtime}
+            id="runtime"
+            type="text"
+            value={runtimeState.value}
+            dispatch={dispatchForm}
+            placeholder={dict.FORM_RUNTIME_PLACEHOLDER}
+            action={FormActionType.UPDATE_RUNTIME}
+            showValidation={showHintsState && !runtimeState.isValid}
+            validationMessage={dict.FORM_RUNTIME_VALIDATION}
+          />
+          <MovieFormTextArea
+            title={dict.FORM_OVERVIEW}
+            id="overview"
+            value={overviewState.value}
+            dispatch={dispatchForm}
+            placeholder={dict.FORM_OVERVIEW_PLACEHOLDER}
+            action={FormActionType.UPDATE_OVERVIEW}
+            showValidation={showHintsState && !overviewState.isValid}
+            validationMessage={dict.FORM_OVERVIEW_VALIDATION}
+          />
+        </div>
+        <div className={styles.control}>
+          <Button
+            className={styles.reset}
+            text="Reset"
+            onClick={() => {
```

Comment:
Same as above about "changeActionCreator".
BTW, it mike sense to move these function to `src/core`

Reply:
In this particular case, no need to pass "action", only the payload - which is actually the end goal for these "actionCreator" function (to only have payload and type hardcoded to a constant)